inet/vibe/inet/url.d

```diff
@@ -419,6 +420,94 @@ struct URL {
 		return cast(NativePath)this.path;
 	}
 
+	/// Decode percent encoded triplets for unreserved or convert to uppercase
+	private string normalize_percent_encoding(const string input)
+	{
+		char[] normalized;
+
+		for (size_t i = 0; i < input.length; i++)
+		{
+			const char c = input[i];
+			if (c == '%')
+			{
+				if (input.length < i + 3)
+					assert(false, "Invalid percent encoding");
+				
+				char conv = cast(char) input[i + 1 .. i + 3].to!ubyte(16);
+				switch (conv)
+				{
+					case 'A': .. case 'Z':
+					case 'a': .. case 'z':
+					case '0': .. case '9':
+					case '-': case '.': case '_': case '~':
+						normalized ~= conv; // Decode unreserved
+						break;
+					default:
+						normalized ~= input[i .. i + 3].toUpper(); // Uppercase HEX
+						break;
+				}
+
+				i += 2;
+			}
+			else
+				normalized ~= c;
+		}
+
+		return normalized.idup;
+	}
+
+	/**
+		Normalizes URL by:
+			- Converting schema and host to lowercase
+			- Removing port if it is the default port for schema
+			- Removing dot segments in path
+			- Converting percent-encoded triplets to uppercase
+			- Adding slash when path is empty
+			- Adding slash to path when path represents a directory
+			- Decoding percent encoded triplets for unreserved characters
+				A-Z a-z 0-9 - . _ ~ 
+
+		Params:
+			isDirectory = Path of the URL represents a directory, a trailing
+				slash is added when true
+	*/
+	void normalize(bool isDirectory = false)
+	{
+		import std.uni : toLower;
+		
+		// Lowercase host and schema
+		this.m_schema = this.m_schema.toLower();
+		this.m_host = this.m_host.toLower();
+
+		// Remove default port
+		if (this.m_port == URL.defaultPort(this.m_schema))
+			this.m_port = 0;
+
+		this.m_path.normalize();
+
+		// Normalize percent encoding, decode unreserved or uppercase hex
+		this.m_queryString = normalize_percent_encoding(this.m_queryString);
+		this.m_anchor = normalize_percent_encoding(this.m_anchor);
+		this.m_path = InetPath(this.m_path.bySegment2.map!(
+				n => InetPath.Segment2.fromTrustedEncodedString(normalize_percent_encoding(n.encodedName))
+			).array);
+
+		// Add trailing slash to empty path
+		if (this.m_path.empty || isDirectory)
+			this.m_path.endsWithSlash = true;		
+	}
+
+	/** Returns the normalized form of the URL.
+
+		See `normalize` for a full description.
+	*/
+	@property URL normalized()
+	const {
```

Comment:
Since this is making a copy and returning that copy, i don't think `@property` is right here.
Note that `@property` is almost useless (the only thing it does is make `typeof(normalized)` be `URL` instead of `URL delegate()`).
I would say, just keep `normalize`, if someone wants to keep the original, it's trivial to write this function from outside the `struct`.